crates/isograph_cli/src/generate_artifacts.rs

```diff
@@ -64,16 +64,123 @@ pub(crate) fn generate_and_write_artifacts(
+        resolver.type_and_field.field_name,
+    );
+    let formatted_field = format!(
+        "entrypoint {}.{}",
+        resolver.type_and_field.type_name, resolver.type_and_field.field_name
+    );
+    s.push_str(&format!(
+        "export function iso<T>(
+            param: T & MatchesWhitespaceAndString<'{}', T>
+        ): IdentityWithParam<typeof entrypoint_{}>;\n",
+        formatted_field,
+        resolver.type_and_field.underscore_separated(),
+    ));
+    (import, s)
+}
+
+fn build_iso_overload_for_schema_resolver(resolver: &ValidatedSchemaResolver) -> (String, String) {
+    let mut s: String = "".to_string();
+    let import = format!(
+        "import {{ ResolverParameterType as field_{} }} from './{}/{}/reader.ts'\n",
+        resolver.type_and_field.underscore_separated(),
+        resolver.type_and_field.type_name,
+        resolver.type_and_field.field_name,
+    );
+    let formatted_field = format!(
+        "field {}.{}",
+        resolver.type_and_field.type_name, resolver.type_and_field.field_name
+    );
+    s.push_str(&format!(
+        "export function iso<T>(
+            param: T & MatchesWhitespaceAndString<'{}', T>
+        ): IdentityWithParam<field_{}>;\n",
+        formatted_field,
+        resolver.type_and_field.underscore_separated(),
+    ));
+    (import, s)
+}
+
+fn build_iso_overload<'schema>(schema: &'schema ValidatedSchema) -> PathAndContent {
+    let mut imports = "".to_string();
+    let mut content = String::from(
+        "type IdentityWithParam<TParam> = <TResolverReturn>(
+    x: (param: TParam) => TResolverReturn
+) => (param: TParam) => TResolverReturn;
+
+type WhitespaceCharacter = ' ' | '\\n';
+type Whitespace<In> = In extends `${WhitespaceCharacter}${infer In}`
+? Whitespace<In>
+: In;
+
+type MatchesWhitespaceAndString<
+TString extends string,
+T
+> = Whitespace<T> extends `${TString}${string}` ? T : never;",
+    );
+    let entrypoints_overload = schema
+        .entrypoints
+        .iter()
+        .map(|resolver_field_id| build_iso_overload_for_entrypoint(*resolver_field_id, schema));
+    let fields_overload = schema
+        .resolvers
+        .iter()
+        .map(|schema_resolver| build_iso_overload_for_schema_resolver(schema_resolver));
+    for (import, field_overload) in fields_overload {
+        imports.push_str(&import);
+        content.push_str(&field_overload);
+    }
+    for (import, entrypoint_overload) in entrypoints_overload {
+        imports.push_str(&import);
+        content.push_str(&entrypoint_overload);
+    }
+    content.push_str(
+        "
+export function iso(_queryText: string): IdentityWithParam<any> {
+  // The name `identity` here is a bit of a double entendre.
+  // First, it is the identity function, constrained to operate
+  // on a very specific type. Thus, the value of b Declare`...`(
+  // someFunction) is someFunction. But furthermore, if one
+  // write b Declare`...` and passes no function, the resolver itself
```

Comment:
Lol this comment is pompous and outdated, and provides no value. Let's remove it lol

Reply:
Fixed with [`951b23a` (#12)](https://github.com/isographlabs/isograph/pull/12/commits/951b23a68cd520ff8d85981ebf3d7e2c280990cb)